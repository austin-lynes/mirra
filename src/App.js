import React, { useState } from "react";


import WelcomeScreen from "./components/welcome-screen"
// MIRRA UI 
import { connect } from "react-redux";


//ROUTER 
import PrivateRoute from './util/privateRoute';
import { Route } from "react-router-dom";
import Dashboard from "./components/dashboard-screen";

//ASYNC STUFF
import { usePromiseTracker } from 'react-promise-tracker';
import Loader from 'react-loader-spinner';
import "react-loader-spinner/dist/loader/css/react-spinner-loader.css"

function App(props) {

  const { auth } = props;
  const { user_loggedIn } = auth;

  const LoadingIndicator = (props) => {

    const { promiseInProgress } = usePromiseTracker();
    
    return (
      promiseInProgress &&
      <div className='loading-indicator'>
        <Loader className='item' type="MutatingDots" color="white" height="100" width="100" />
      </div>
    )
  
  }

  return (
    <div data-testid='view' className="view">
      <div className="drag-bar" />
      <LoadingIndicator />
      <Route exact path='/' component={WelcomeScreen} />
      <PrivateRoute
        path='/dashboard'
        component={Dashboard} />

    </div>
  );
}


const _props = (store) => {
  return {
    auth: store.auth
  }
}
const _actions = {

}

export default connect(_props, _actions)(App);
