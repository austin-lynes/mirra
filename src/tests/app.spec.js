import React from 'react';
import { render } from '@testing-library/react';
import { shallow } from 'enzyme';
import { toBeInTheDocument } from '@testing-library/jest-dom';

//#region  MOCK
import App from '../App';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware } from 'redux';
import logger from 'redux-logger';

const mockReducer = (state = { auth: {}, }, action) => {
    switch (action.payload) {
        default:
            return state;
    }
}

const MockApp = () => {

    const test_store = createStore(mockReducer, applyMiddleware(logger));

    return (
        <BrowserRouter>
            <Provider store={test_store}>
                <App />
            </Provider>
        </BrowserRouter>
    )
}
//#endregion

describe('src/app', () => {
    it('renders without crashing', () => {
        const view = shallow(<MockApp />);
        expect(view).toMatchSnapshot();
    })

    it('it has two buttons', () => {
        const { getAllByTestId } = render(<MockApp />);
        const buttons = getAllByTestId(/button/i);
        expect(buttons.length).toBe(2);
    })

    it('it has a welcome banner', () => {
        const { getByTestId } = render(<MockApp />);
        const welcome_msg = getByTestId(/welcome-banner/i);
        expect(welcome_msg.textContent).toBe("Welcome to Mirra!");
    })

})