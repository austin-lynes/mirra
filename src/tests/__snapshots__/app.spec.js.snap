// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/app renders without crashing 1`] = `
<BrowserRouter>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <Connect(App) />
  </Provider>
</BrowserRouter>
`;
